### PR TITLE
feat: add a description for sdm

### PR DESCRIPTION
**Pull request by DenisKolodin**
Description
---
The description for the sdm crate and changes required for the UI.

Motivation and Context
---

How Has This Been Tested?
---

### CODE REVIEW

File: libs/README.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    Progress(TaskProgress),
+    Active,
+}
+```
+
+And to get to know the container is ready or not (in progress) we should
+get a value of the following field in the frontend:
+
+```rust
+state.containers["Base Node"].status
```
Comment (shanimal08) [inline libs/README.md:336]: is there a list of all the container names like this we use to get these states, or just all based off the `display_name` [here](https://github.com/tari-project/tari-launchpad/blob/5c92aef886470c30cefcb8617c9e4ffc9a76ffbe/backend/src/docker/models.rs#L224)?

File: libs/README.md [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+
+The obvious solution to this problem is: let's just take into account
+all these cases in the frontend part. But how to test all that? To cover
+that cases with tests, we have to create a lot of fragile tests that will
+break simply because we move the buttons to other places.
+
+Another potential problem is the destruction of the main window since
+the frontend part controls the backend. If the main window is destroyed
+accidentally by the system, we lose the state, and the containers
+become unmanaged.
```
Comment (CjS77) [inline libs/README.md:123]: There is a call back when the window closes that allows the backend to clean up

File: libs/README.md [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+
+It is also impossible to create a multi-window interface, or
+a container that manages others (a kind of launchpad for servers).
+
+Another class of potential problems is related to the fact that we,
+for example, send SQL queries directly from the frontend to the backend.
+And, in theory, we have to cover that requests with tests as well.
```
Comment (CjS77) [inline libs/README.md:130]: Note that The SQL queries are for persisting state for the front-end and are not strictly related to the management of containers.

File: libs/README.md [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+
+And to send a control action we should send an event to the `tari://actions`
+stream:
+
+```js
+function sendEvent(action) {
+    emit('tari://actions', action)
+}
+```
+
```
Comment (CjS77) [inline libs/README.md:173]: So you're suggesting that the FE must become a Finite State Machine that reacts to events from the `reactions` stream, and can request state changes via the `actions` stream?

File: libs/README.md [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+The state above should be declared in the frontend part and received from the
+`tari://reactions` stream that send events of the following `Reaction` type:
+
+```rust
+pub enum Reaction {
+    State(LaunchpadState),
+    Delta(LaunchpadDelta),
+}
```
Comment (CjS77) [inline libs/README.md:209]: Ok, so the first message (or when re-requested) would be a `State` message to set the state. Subsequent messages will mostly be `Delta` messages which the FE FSM can use to decide what UI changes to make and/or which actions to send.

There are some further elaborations below which elucidate a bit more.

**Review by CjS77 — APPROVED** [review]
LGTM - made a few comments, but I think this will be very helpful, thanks

**Comment by CjS77** [comment]
@shanimal08 once you comment with a standalone ACK, this will merge automatically.

File: libs/README.md [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+### Events channels
+
+To connect UI to the new engine we should use two calls on the frontend that is
+the part of the Tauri API.
+
+To subscribe to the state and changes of that we should subscribe to the
+`tari://reactions` stream of events:
+
+```js
+const unlisten = await listen('tari://reactions', (event) => { })
```
Comment (shanimal08) [inline libs/README.md:162]: this needs to be updated if the changes in #194 aren't added

**Review by shanimal08 — APPROVED** [review]
ACK

**Comment by CjS77** [comment]
ACK

**Pull request merged**